include/aikido/rviz/TrajectoryMarker.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef AIKIDO_RVIZ_TRAJECTORYMARKER_HPP_
+#define AIKIDO_RVIZ_TRAJECTORYMARKER_HPP_
+
+#include <dart/dynamics/Frame.hpp>
+#include <interactive_markers/interactive_marker_server.h>
+#include <visualization_msgs/InteractiveMarker.h>
+#include "aikido/trajectory/Trajectory.hpp"
+
+namespace aikido {
+namespace rviz {
+
+/// A wrapper class of RViz InteractiveMarker for visualizing AIKIDO trajectory
+/// in RViz.
+class TrajectoryMarker final
+{
+public:
+  /// Constructor.
+  /// \param[in] markerServer RViz marker server.
+  /// \param[in] frameId RViz frame ID.
+  /// \param[in] trajectory C-sapce (or joint-space) trajectory.
+  /// \param[in] skeleton Target DART meta skeleton that the C-space trajectory
+  /// will be applied to compute the visualizing task-space trajectory.
+  /// \param[in] frame Target DART frame where the trajectory of its origin
+  /// will be visualized.
+  /// \param[in] rgba Color and alpha of the visualizing task-space trajectory.
+  /// Default is [RGBA: 0.75, 0.75, 0.75, 0.75].
+  /// \param[in] thickness Thickness of the visualizing task-space trajectory.
+  /// Default is 0.01.
+  /// \param[in] numLineSegments Number of line-segments of the visualizing
+  /// task-space trajectory. Default is 16.
+  TrajectoryMarker(
+      interactive_markers::InteractiveMarkerServer* markerServer,
+      const std::string& frameId,
+      trajectory::ConstTrajectoryPtr trajectory,
+      const dart::dynamics::MetaSkeleton& skeleton,
+      const dart::dynamics::Frame& frame,
+      const Eigen::Vector4d& rgba = Eigen::Vector4d::Constant(0.75),
+      double thickness = 0.01,
+      std::size_t numLineSegments = 16u);
+
+  /// Destructor
+  ~TrajectoryMarker();
+
+  /// Sets or updates trajectory to visualize
```

Comment:
From reading the code, it seems that passing in `nullptr` will clear the current trajectory. Maybe we can explain that here as well?